src/test/app/Ticket_test.cpp

```diff
@@ -0,0 +1,419 @@
+    /// each meta node entry. When the transaction was a cancel with differing
+    /// target and account, there will be 4 complete items, otherwise the last
+    /// entry will be an empty object
+    auto
+    checkTicketMeta(
+        test::jtx::Env& env,
+        bool other_target = false,
+        bool expiration = false)
+    {
+        using namespace std::string_literals;
+        auto const& tx = env.tx ()->getJson (0);
+        bool is_cancel = tx[jss::TransactionType] == "TicketCancel";
+
+        auto const& jvm = env.meta ()->getJson (0);
+        std::array<Json::Value, 4> retval;
+
+        // these are the affected nodes that we expect for
+        // a few different scenarios.
+        // tuple is index, field name, and label (LedgerEntryType)
+        std::vector<
+            std::tuple<std::size_t, std::string, std::string>
+        > expected_nodes;
+
+        if (is_cancel && other_target)
+        {
+            expected_nodes = {
+                std::make_tuple(0, sfModifiedNode.fieldName, "AccountRoot"s),
+                std::make_tuple(
+                    expiration ? 2: 1, sfModifiedNode.fieldName, "AccountRoot"s),
+                std::make_tuple(
+                    expiration ? 1: 2, sfDeletedNode.fieldName, "Ticket"s),
+                std::make_tuple(3, sfDeletedNode.fieldName, "DirectoryNode"s)
+            };
+        }
+        else
+        {
+            expected_nodes = {
+                std::make_tuple(0, sfModifiedNode.fieldName, "AccountRoot"s),
+                std::make_tuple(1,
+                    is_cancel ?
+                        sfDeletedNode.fieldName : sfCreatedNode.fieldName,
+                    "Ticket"s),
+                std::make_tuple(2,
+                    is_cancel ?
+                        sfDeletedNode.fieldName : sfCreatedNode.fieldName,
+                 "DirectoryNode"s)
+            };
+        }
+
+        BEAST_EXPECT(jvm.isMember (sfAffectedNodes.fieldName));
+        BEAST_EXPECT(jvm[sfAffectedNodes.fieldName].isArray());
+        BEAST_EXPECT(
+            jvm[sfAffectedNodes.fieldName].size() == expected_nodes.size());
+
+        // verify the actual metadata against the expected
+        for (auto const& it : expected_nodes)
+        {
+            auto const& idx = std::get<0>(it);
+            auto const& field = std::get<1>(it);
+            auto const& type = std::get<2>(it);
+            BEAST_EXPECT(jvm[sfAffectedNodes.fieldName][idx].isMember(field));
+            retval[idx] = jvm[sfAffectedNodes.fieldName][idx][field];
+            BEAST_EXPECT(retval[idx][sfLedgerEntryType.fieldName] == type);
+        }
+
+        return retval;
+    }
+
+    void testTicketNotEnabled ()
+    {
+        testcase ("Feature Not Enabled");
+
+        using namespace test::jtx;
+        Env env {*this};
+
+        env (ticket::create (env.master), ter(temDISABLED));
+
+        auto jv = ticket::create (env.master);
```

Comment:
I made https://github.com/bachase/rippled/commit/bebbd2c5edc6ec72baf63da8ac19b49d67c39fe8 if you want to cherrypick.